services/QuillLMS/app/controllers/accounts_controller.rb

```diff
@@ -39,26 +38,25 @@ def create
     end
   end
 
-  def update
-    user_params.delete(:password) unless user_params[:password].present?
-    @user = current_user
+  def edit
+    @user = User.find_by_token(params[:id])
+    return if @user.present?
 
-    if user_params[:username] == @user.username
-      validate_username = false
-    else
-      validate_username = true
-    end
+    redirect_to profile_path, notice: "Sorry, this link has expired. Please contact your Quill admin or the <a href='mailto:hello@quill.org'>Quill support team</a>".html_safe
```

Comment:
I like to refactor error messages into [en.yml](https://github.com/empirical-org/Empirical-Core/blob/0d83fb606899718411591c1ee25dea9d44b887fb/services/QuillLMS/config/locales/en.yml#L22) for  several reasons:  
1)  Single source of truth with the text (e.g. simplifies [testing](https://github.com/empirical-org/Empirical-Core/pull/10020/files#diff-5ef8a8db0bcd0f2523214ca8da214420110b48536a0bb4a7b1d7ce84287e4a6bR183))
2)  Sets up a nice pattern for i18n later.
3)  Keeps lines short `redirect_to profile_path, notice: t('accounts.edit.expired_link')`
4)  It's easier to keep a consistent voice/theme with messages if they're all in the same place.